nengo_bones/scripts/check_bones.py

```diff
@@ -0,0 +1,61 @@
+"""Applies validation to auto-generated files."""
+
+import os
+import sys
+
+import click
+
+import nengo_bones
+
+
+@click.command()
+@click.option("--root-dir", default=".",
+              help="Directory containing files to be checked")
+def main(root_dir):
+    """
+    Validates auto-generated project files.
+
+    Notes
+    -----
+    This does not check the ci scripts, because those are generated on-the-fly
+    in TravisCI (so any ci files we did find here are likely local artifacts).
+    """
+
+    print("*" * 50)
+    print("Checking version of nengo-bones generated files:")
+    print("root dir: %s\n" % root_dir)
+
+    passed = True
+
+    for filename in [".travis.yml", ".codecov.yml"]:
+        print(filename + ":")
+
+        if not os.path.exists(os.path.join(root_dir, filename)):
+            print("  File not found")
+            continue
```

Comment:
Why is it okay to continue if the file doesn't exist?

Reply:
My thinking was that not all repos might want to use all of the templated files.  E.g. for something like `nengo-sphinx-theme`, maybe coverage reports don't make much sense, so they wouldn't be using `.codecov.yml`.

Reply:
Could this cause any weird issues? If for example I accidentally deleted one of these two files? Or would it fail predictably later on?

Reply:
For `.travis.yml` you wouldn't get any Travis builds running, so that would be pretty obvious.  For `.codecov.yml` it'll just fall back on default settings, so it wouldn't be as obvious.  Although `.codecov.yml` only affects the settings once it's committed to the repo, so in order for that to happen you'd have to accidentally delete/move the file and then also commit that change (and presumably have that pass some code review).

We could also add a `--strict` flag if that still seems problematic, which return non-zero status if any of the expected files aren't found with the right version.  Although we'd then have to decide whether we want that to be the default when running on TravisCI or not.

Reply:
Another possibility later on could be to have the `.nengo-bones.yml` file explicitly opt-out of the file if it shouldn't be there. This could also be used as a safe-guard to prevent someone from accidentally generating one. But given that it's innocuous at the moment I'll mark this resolved for now.

Reply:
Actually would it make sense to use the existence of `codecov_config` and `travis_config` in `.nengobones.yml` to determine whether or not the file should exist?

Reply:
Yeah that'd make sense I think.

Reply:
I added a couple tests for the current behaviour and added a TODO to improve this in the future. It's currently non-trivial because `check_bones` does not have immediate access to the `config`.